tsdb/engine/tsm1/engine.go

```diff
@@ -1893,8 +1893,22 @@ func (e *Engine) WriteSnapshot() (err error) {
 // CreateSnapshot will create a temp directory that holds
 // temporary hardlinks to the underylyng shard files.
 func (e *Engine) CreateSnapshot() (string, error) {
-	if err := e.WriteSnapshot(); err != nil {
-		return "", err
+	var err error
```

Comment:
Question - is this only used for backups?

Reply:
For what I can tell this is the unique place where we use it

https://github.com/influxdata/influxdb/blob/a56e0226e61bbcfca88b1e0012e27463d10c0fe4/storage/engine.go#L674

So yes

Reply:
I think `CreateSnapshot()` is called from:

https://github.com/influxdata/influxdb/blob/1.8/tsdb/shard.go#L1125
https://github.com/influxdata/influxdb/blob/1.8/tsdb/engine/tsm1/engine.go#L981
https://github.com/influxdata/influxdb/blob/1.8/tsdb/engine/tsm1/engine.go#L915